src/components/chip/chip.tsx

```diff
@@ -0,0 +1,279 @@
+            >
+                {this.renderIcon()}
+                {this.renderLabel()}
+                {this.renderBadge()}
+            </button>,
+            this.renderRemoveButton(),
+        ];
+    };
+
+    private renderAsLink = () => {
+        return [
+            <a
+                class="chip"
+                href={this.link.href}
+                title={this.link.title}
+                aria-disabled={this.disabled || this.readonly}
+                tabindex={this.disabled ? -1 : 0}
+                onClick={(e) => {
+                    if (this.disabled) {
+                        e.preventDefault();
+                    }
+                }}
+                onKeyDown={this.handleDeleteKeyDown}
+            >
+                {this.renderIcon()}
+                {this.renderLabel()}
+                {this.renderBadge()}
+            </a>,
+            this.renderRemoveButton(),
+        ];
+    };
+
+    private renderLabel = () => {
+        return <span class="text">{this.text}</span>;
+    };
+
+    private renderIcon() {
+        const icon = getIconName(this.icon);
+
+        if (!icon) {
+            return;
+        }
+
+        return (
+            <limel-icon
+                badge={true}
+                name={icon}
+                style={{
+                    color: `${(this.icon as Icon)?.color}`,
+                    'background-color': `${(this.icon as Icon)
+                        ?.backgroundColor}`,
+                }}
+            />
+        );
+    }
+
+    private renderBadge() {
+        if (!this.badge) {
+            return;
+        }
+
+        return <limel-badge label={this.badge} />;
+    }
+
+    private renderRemoveButton() {
+        if (!this.removable || this.readonly || this.disabled) {
+            return;
+        }
+
+        const svgData =
+            '<svg viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg"><path fill="none" stroke="currentColor" stroke-width="2" d="m8 8 16 16M24 8 8 24"/></svg>';
+
+        return (
+            <button
+                class="trailing-button remove-button"
+                tabIndex={-1}
+                aria-label={this.removeChipLabel}
+                innerHTML={svgData}
+                onClick={this.handleRemoveClick}
+            />
+        );
+    }
+
+    private handleRemoveClick = (event: MouseEvent | KeyboardEvent) => {
+        event.stopPropagation();
+        this.remove.emit(this);
+    };
+
+    private handleDeleteKeyDown = (event: KeyboardEvent) => {
+        if (!this.removable) {
+            return;
+        }
+
+        const keys = [DELETE, BACKSPACE];
+        const keycodes = [DELETE_KEY_CODE, BACKSPACE_KEY_CODE];
+
+        if (keys.includes(event.key) || keycodes.includes(event.keyCode)) {
+            this.handleRemoveClick(event);
+        }
+    };
+
+    private removeEnterClickable() {
+        const element = this.button ?? this.anchor;
+        removeEnterClickable(element);
+    }
```

Comment:
this was just a copy/paste from "action bar item" if I'm not hallucinating. 🤔  If this is a problem, it should probably be fixed in several places.

Reply:
by the way, just some clarifications, we only want to prevent repeatedly triggered actions when using the <kbd>Enter</kbd> on keyboard, when the host is clicked. Because the consumer will add the `onClick` on the host element, even though the component contains a `<button>` element in itself.

However, when the component renders a hyperlink (an `<a>` element), there is hopefully no onClick defined.

Reply:
I think we are leaking memory here, because I don't think the listeners we set with `makeEnterClickable` is ever removed, since the two functions are called on different elements. That's probably a problem in breadcrumbs too then, yes.

Reply:
@LawrenceBorst you know this best

Reply:
I think it needs to be `this.host` as well :), but I'll play around with it. Maybe moving forward we should give the consumer a random `id` which they own with which they can destruct the object in the service, and use information hiding to avoid other consumers peeking around for other ids.

Reply:
⚡